File: web/typescript-cucumber-cypress/cypress/e2e/steps/LoginPageDefinitions.ts
import { Given, Then, When } from "@badeball/cypress-cucumber-preprocessor";
import LoginPageActions from "../pages/LoginPageActions";
import HomePageActions from "../pages/HomePageActions";
import { passwordStep, usernameStep } from "./SignupPageDefinitions";

const objLoginPage = new LoginPageActions;
const objHomePage = new HomePageActions;

Given("El usuario está en la página PRODUCT STORE {string}", (url: string) => {
   cy.visit(url);
})

When("El usuario ingresa el nombre de usuario como {string} y la contraseña como {string}", (usuario: string, clave: string) => {
    objHomePage.clickBotonLogin();
    objLoginPage.typeTextoUsuario(usuario);
    objLoginPage.typeTextoClave(clave);
    objLoginPage.clickBotonLogin();
})

When("El usuario inicia sesion ingresado el nombre de usuario como {string} y la contraseña como {string}",(username: string, password: string) => {
    objHomePage.clickBotonLogin();
    objLoginPage.typeTextoUsuario(username);
    objLoginPage.typeTextoClave(password);
    objLoginPage.clickBotonLogin();

    cy.wait(3000);
})

Then("El usuario debería poder ver su usuario {string}", (username: string) => {
    cy.wait(3000);
    objHomePage.getUsuario().should('contain', "Welcome " + username);
    cy.screenshot({capture : 'runner'})
})

Then("El usuario debería poder ver el mensaje de error {string}", (msg: string) => {
    cy.on('window:alert', (message: string) => {
        expect(message).to.equal(msg);
    });
    cy.wait(2000);
})

Then("El usuario debería poder ver un mensaje {string}", (msg: string) => {
    cy.on('window:alert', (message: string) => {
        expect(message).to.equal(msg);
    });
    objLoginPage.clickBotonLogin();
})

Then("El usuario inicia sesion", () => {
    objHomePage.clickBotonLogin();
    objLoginPage.typeTextoUsuario(usernameStep);
    objLoginPage.typeTextoClave(passwordStep);
    objLoginPage.clickBotonLogin();
})

File: web/java-maven-cucumber-selenium-testng/src/test/java/com/jcst/definitions/LoginPageDefinitions.java
package com.jcst.definitions;

import com.jcst.actions.HomePageActions;
import com.jcst.actions.LoginPageActions;
import com.jcst.utils.HelperClass;
import com.jcst.utils.HelperUtils;
import io.cucumber.java.en.And;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.testng.Assert;

public class LoginPageDefinitions {

    HomePageActions objHomePage = new HomePageActions();
    LoginPageActions objLoginPage = new LoginPageActions();

    @Given("El usuario está en la página PRODUCT STORE {string}")
    public void elUsuarioEstaEnLaPaginaPRODUCTSTORE(String url) {
        HelperClass.openPage(url);
    }

    @When("El usuario ingresa el nombre de usuario como {string} y la contraseña como {string}")
    public void elUsuarioIngresaElNombreDeUsuarioComoYLaContrasenaComo(String username, String password) {
        objHomePage.clickBotonLogin();
        objLoginPage.typeTextUsuario(username);
        objLoginPage.typeTextClave(password);
        objLoginPage.clickBotonLogin();
    }

    @When("El usuario inicia sesion ingresado el nombre de usuario como {string} y la contraseña como {string}")
    public void elUsuarioIniciaSesionIngresadoElNombreDeUsuarioComoYLaContrasenaComo(String username, String password) {
        objHomePage.clickBotonLogin();
        objLoginPage.typeTextUsuario(username);
        objLoginPage.typeTextClave(password);
        objLoginPage.clickBotonLogin();
    }

    @Then("El usuario debería poder ver su usuario {string}")
    public void elUsuarioDeberiaPoderVerSuUsuario(String username) {
        String usuarioObtenido = objHomePage.getUsuario();
        Assert.assertEquals(usuarioObtenido, "Welcome " + username);
    }

    @Then("El usuario debería poder ver el mensaje de error {string}")
    public void elUsuarioDeberiaPoderVerElMensajeDeError(String message) {
        String mensajeObtenido = objLoginPage.alertaError();
        Assert.assertEquals(mensajeObtenido, message);
    }

    @Then("El usuario debería poder ver un mensaje {string}")
    public void elUsuarioDeberiaPoderVerUnMensaje(String message) {
        String mensajeObtenido = objLoginPage.alertaError();
        Assert.assertEquals(mensajeObtenido, message);
    }

    @And("El usuario inicia sesion")
    public void elUsuarioIniciaSesion() {
        objHomePage.clickBotonLogin();
        objLoginPage.typeTextUsuario(SignupPageDefinitions.usernameStep);
        objLoginPage.typeTextClave(SignupPageDefinitions.passwordStep);
        objLoginPage.clickBotonLogin();
        HelperUtils.sleep(1);
    }
}



